.github/workflows/internal-ci-checks.yml

```
@@ -0,0 +1,28 @@
+name: Internal CI Checks
+# This workflows checks if the author of a PR is a member of the CodeQL team
+# and adds `ready-for-internal-ci` label to trigger the internal CI
+
+on:
+  pull_request_target:
+
+jobs:
+  set-label:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set a label to trigger internal CI checks
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          USERNAME="${{ github.event.pull_request.user.login }}"
+          LABEL="ready-for-internal-ci"
+          set +eo pipefail
+          echo "Checking if user $USERNAME is a member of the CodeQL team"
+          gh api -H "Accept: application/vnd.github+json" /orgs/github/teams/codeql/memberships/$USERNAME > /dev/null 2>&1
+          if [ "$?" == 0 ]; then
```

We can drop `-H "Accept: application/vnd.github+json"` . It works fine without, I suppose the `gh api` command sets that by default. I'd recommend adding quotes around the URL argument even though the `USERNAME` variable is likely free of spaces and other unwanted symbols.

The API returns a bit of JSON like  `{"role":"member","state":"active","url":...}` . I think we should check the `state` property and accept only `state: "active"`. The state property can also have value `pending` and perhaps even more values. A `--jq`  flag of `gh api` is very handy.

```suggestion          
          if [ "$(gh api "/orgs/github/teams/codeql/memberships/$USERNAME" --jq .state)" == "active" ]; then
```